ingestion-lambda/src/categoryCodes.ts

```diff
@@ -110,3 +112,32 @@ export function processUnknownFingerpostCategoryCodes(
 
 	return deduped;
 }
+
+export async function inferRegionCategoryFromText(content: string | undefined): Promise<string | undefined> {
+	if (!content) {
+		return undefined;
+	}
+
+	const { default: nlp } = await import('compromise');
+
+	const doc = nlp(content, lexicon) as {
+		places: () => { out: (format: string) => unknown };
+	};
+
+	const rawPlaces = doc.places().out('array');
+
+	if (!Array.isArray(rawPlaces)) {
+		return undefined;
+	}
+
+	const places = (rawPlaces as string[])
+		.flatMap((place) => place.split(/[,\n]/))
+		.map((place) => place.trim().toLowerCase())
+		.filter((place) => !!place && place.length > 0);
+
+	const isUk = places.some((place) =>
+		ukPlaces.some((ukPlace) => place.includes(ukPlace)),
```

Comment:
Matching on `includes` means that the following test case fails:

```
it('should return undefined when only non-UK places are mentioned', async () => {
	const content = 'Kingsford Smith, Bradfield, Sydney, and Warringah in New South Wales and Fadden in Queensland are the five most expensive electorates, data shows.';
	expect(await inferRegionCategoryFromText(content)).toEqual(undefined);
});
```

Could we try doing exact match, or are there other edge cases that we'd get wrong if we do that? I know there will always be edge cases for us here, so not blocking 👍

Reply:
It looks like an edge case to keep in mind. I was thinking to try out this iteration and make further updates to it later if needed.